pkg/universe.dagger.io/aws/aws.cue

```diff
@@ -7,18 +7,24 @@ import (
 	"universe.dagger.io/docker"
 )
 
-#DefaultLinuxVersion: "amazonlinux:2.0.20220121.0@sha256:f3a37f84f2644095e2c6f6fdf2bf4dbf68d5436c51afcfbfa747a5de391d5d62"
-#DefaultCliVersion:   "2.4.12"
+_#DefaultLinuxName:       "amazonlinux"
+_#DefaultLinuxVersion:    "2.0.20220121.0@sha256:f3a37f84f2644095e2c6f6fdf2bf4dbf68d5436c51afcfbfa747a5de391d5d62"
+_#DefaultLinuxRepository: "index.docker.io"
+#DefaultCliVersion:       "2.4.12"
 
 // Build provides a docker.#Image with the aws cli pre-installed to Amazon Linux 2.
 // Can be customized with packages, and can be used with docker.#Run for executing custom scripts.
 // Used by default with aws.#Run
 #Build: {
 
+	name:       *_#DefaultLinuxName | string
+	repository: *_#DefaultLinuxRepository | string
+	version:    *_#DefaultLinuxVersion | string
+
 	docker.#Build & {
 		steps: [
 			docker.#Pull & {
-				source: #DefaultLinuxVersion
+				source: "\(repository)/\(name):\(version)"
```

Comment:
Same as my comment in https://github.com/dagger/dagger/pull/2666#pullrequestreview-1011413983, I do not understand the benefit of those changes?
If you want to provide a custom image from another repository, just do it using `input` field.

Reply:
Yes, you're right 😬